autogen/agentchat/contrib/capabilities/generate_images.py

```diff
@@ -0,0 +1,156 @@
+
+SYSTEM_MESSAGE = "You've been given the special ability to generate images."
+DESCRIPTION_MESSAGE = "This agent has the ability to generate images."
+
+
+class ImageGenerator(abc.ABC):
+    @abc.abstractmethod
+    def generate_image(self, prompt: str) -> Image:
+        raise NotImplementedError()
+
+
+class DalleImageGenerator(ImageGenerator):
+    def __init__(
+        self,
+        llm_config: Dict,
+        resolution: Literal["256x256", "512x512", "1024x1024", "1792x1024", "1024x1792"] = "1024x1024",
+        quality: Literal["standard", "hd"] = "standard",
+        num_images: int = 1,
+    ):
+        config_list = llm_config["config_list"]
+        _validate_dalle_model(config_list[0]["model"])
+        _validate_resolution_format(resolution)
+
+        self._model = config_list[0]["model"]
+        self._resolution = resolution
+        self._quality = quality
+        self._num_images = num_images
+        self._dalle_client = OpenAI(api_key=config_list[0]["api_key"])
+
+    def generate_image(self, prompt: str) -> Image:
+        """Got this from Beibin Li's work in agentchat_dalle_and_gpt4v.ipynb"""
+        # Function implementation...
+        cache = Cache(".cache/")  # Create a cache directory
+        key = (self._model, prompt, self._resolution, self._quality, self._num_images)
+
+        # if key in cache:
+        #     return img_utils._to_pil(cache[key])
+
+        # If not in cache, compute and store the result
+        response = self._dalle_client.images.generate(
+            model=self._model,
+            prompt=prompt,
+            size=self._resolution,
+            quality=self._quality,
+            n=self._num_images,
+        )
+        image_url = response.data[0].url
+        img_data = img_utils.get_image_data(image_url)
+        cache[key] = img_data
+        return img_utils._to_pil(img_data)
+
+
+class ImageGeneration(AgentCapability):
+    def __init__(
+        self, image_generator: ImageGenerator, text_analyzer_llm_config: Optional[Dict] = None, verbosity: int = 0
+    ):
+        self._image_generator = image_generator
+        self._text_analyzer_llm_config = text_analyzer_llm_config
+        self._verbosity = verbosity
+
+        self._agent: Optional[ConversableAgent] = None
+        self._text_analyzer: Optional[TextAnalyzerAgent] = None
+
+    def add_to_agent(self, agent: ConversableAgent):
+        self._agent = agent
+
+        agent.register_reply([Agent, None], self._image_gen_reply, position=2)
+
+        self._text_analyzer_llm_config = self._text_analyzer_llm_config or agent.llm_config
+        self._text_analyzer = TextAnalyzerAgent(llm_config=self._text_analyzer_llm_config)
+
+        agent.update_system_message(agent.system_message + "\n" + SYSTEM_MESSAGE)
+        # Hack to update the description of the agent, useful when used in a group chat
+        agent.description += "\n" + DESCRIPTION_MESSAGE
+
+    def _image_gen_reply(
+        self,
+        reciepient: ConversableAgent,
```

Comment:
For future reference, we need to check the recipient for multi-modality capability. @BeibinLi